File: backend/ScheduleCore/Models/DTO/ResponseData.cs
namespace ScheduleCore.Models.ViewModels;

public class ResponseData
{
    public List<CursoHorarioDTO> Horario { get; set; }
    public List<CursoConAdvertencia> Advertencias { get; set; }
    public List<CarreraColor> Carreras { get; set; }
    public TimeOnly HoraInicio { get; set; }
    public TimeOnly HoraFin { get; set; }
    public int DuracionPeriodo { get; set; }
}

public class CarreraColor
{
    public int CodigoCarrera { get; set; }
    public string Nombre { get; set; }
    public string Color { get; set; }
}

File: frontend/src/app/models/response-data.ts
import { Carrera } from "./carrera";
import { CursoAdvertencia } from "./curso-advertencia";
import { CursoHorarioDto } from "./curso-horario-dto";
import { Salon } from "./salon";

export interface ResponseData {
    horario: CursoHorarioDto[];
    advertencias: CursoAdvertencia[];
    carreras: Carrera[];
    salones : Salon[];
    horaInicio: string;
    horaFin: string;
    duracionPeriodo: number;
    porcentajeEfectividad: number;
}
